torch/csrc/autograd/function.h

```diff
@@ -190,7 +190,8 @@ struct TORCH_API Node : std::enable_shared_from_this<Node> {
       bool is_tensor_subclass) noexcept {
     // NOLINTNEXTLINE(cppcoreguidelines-init-variables)
     uint32_t input_nr = input_metadata_.size();
-    input_metadata_.emplace_back(options, shape, is_tensor_subclass);
```

Comment:
This constructor is only used here: https://github.com/drisspg/pytorch/blob/59fb6b4622484be3af1ba15172d898f824f4263f/torch/csrc/autograd/variable.cpp#L657

The hot path where this is called is when self is a view of base have a different versions, I.e. an inplace op has been called on self which itself is a view of base.  This normally would get a node: AsStridedBackward0.  

Looking at AsStridedBackward0 there are many calls to sizes which would not work for nested tensors. Also there are now view ops currently implemented for nested tensors. SO I thought it was okay to not handle this scenario in this constructor.  

- Should a comment be made in `variable.cpp` to this effect?
- Should a `TORCH_CHECK(!self.is_nested())` be added

If and when views on nested tensors is ultimately supported the plan would be to update this input_metadata constructor to take in 2 tensors -> self, base. And then add another input_metadata constructor to handle this case when creating the input_metadata. This would effectively quarantine the handling of "nested ness" to metadata.h

Reply:
cc @albanD

Reply:
I think a better way to fix this would be to make nested Tensor record view_fn instead of using as_strided.
I don't remember where is the switch to change that though, do you @soulitzer ?

Reply:
Looks like as long as the nested tensor's tensorImpl does not support as_strided, the inplace or view kernel would set the view fn. 
https://github.com/pytorch/pytorch/blob/88205886d72486fbeac420c7df99bfe460c38436/tools/autograd/gen_inplace_or_view_type.py#L151-L158

Reply:
Ho right `support_as_strided()`! Perfect!
@drisspg  can you make sure this is properly returning False for NestedTensor? That should solve your issue here.

Reply:
https://github.com/pytorch/pytorch/pull/79616